File: 0283-move-zeroes/0283-move-zeroes.go
func moveZeroes(nums []int)  {
    left:=0
    
    for right := 0; right<len(nums); right++ {
        if nums[right] != 0 && nums[left] ==0 {
            nums[left] = nums[right]
            nums[right] = 0
        }
        
        if nums[left] != 0 {
            left ++
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        left = 0
        right = 0
        
        while right < len(nums):
            
            if nums[left] ==0 and nums[right] != 0:
                nums[left] = nums[right]
                nums[right] = 0
            
            if nums[left] != 0:
                left += 1
            right += 1
                
        